src/Symfony/Component/Lock/Store/MongoDbStore.php

```diff
@@ -0,0 +1,263 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Lock\Store;
+
+use Symfony\Component\Lock\Exception\InvalidArgumentException;
+use Symfony\Component\Lock\Exception\LockConflictedException;
+use Symfony\Component\Lock\Exception\LockExpiredException;
+use Symfony\Component\Lock\Exception\LockStorageException;
+use Symfony\Component\Lock\Exception\NotSupportedException;
+use Symfony\Component\Lock\Key;
+use Symfony\Component\Lock\StoreInterface;
+
+/**
+ * MongoDbStore is a StoreInterface implementation using MongoDB as store engine.
+ *
+ * @author Joe Bennett <joe@assimtech.com>
+ */
+class MongoDbStore implements StoreInterface
+{
+    private $mongo;
+    private $options;
+    private $initialTtl;
+    private $collection;
+
+    /**
+     * @param \MongoDB\Client $mongo
+     * @param array           $options
+     *
+     * database:        The name of the database [required]
+     * collection:      The name of the collection [default: lock]
+     * drift:           Seconds to extend expiries to account for clock discrepancies [default: 0.0]
+     *
+     * Please be aware, MongoDbStore relies on clock syncronisation between
+     * the application nodes and mongodb servers. The drift option can be used
+     * to extend expiries to account for clock discrepancies.
+     *
+     * A TTL index MUST BE used on MongoDB 2.2+ to automatically clean up expired locks.
+     *
+     *     db.lock.ensureIndex(
+     *         { "expires_at": 1 },
+     *         { "expireAfterSeconds": 0 }
+     *     )
+     *
+     * @see http://docs.mongodb.org/manual/tutorial/expire-data/
+     *
+     * writeConcern, readConcern and readPreference are not specified by MongoDbStore
+     * meaning the collection's settings will take effect.
+     * @see https://docs.mongodb.com/manual/applications/replication/
+     *
+     * Please note, the Symfony\Component\Lock\Key's $resource
+     * must not exceed 1024 bytes including structural overhead.
+     * @see https://docs.mongodb.com/manual/reference/limits/#Index-Key-Limit
+     *
+     * @param float $initialTtl The expiration delay of locks in seconds
```

Comment:
Why `$initialTtl` is a constructor argument, whereas every other options are passed into `$options`?

Reply:
I'll move it to `$options`.

I've been thinking it would be simpler to remove the `drift` option in favour of asking the developer set a slightly higher lockTTL. This would both simplify the implementation and probably make more sense to the developer rather than trying to explain why drift is a separate option. I would have to document the danger of clock drift against this Store. What are your thoughts?